test/SchedulerTest.cpp

```diff
@@ -74,3 +107,41 @@ TYPED_TEST(SchedulerTest, ScheduledChannelCreationWithTypes) {
   EXPECT_EQ(sc.GetCapacity(), 1);
   EXPECT_EQ(sc.GetPacketSize(), sizeof(pipert::Packet<TypeParam>));
 }
+
+TEST(Scheduler, SchedulerPipeLineTest) {
+  pipert::Scheduler sch;
+  int channel_capacity = 10;
+
+  pipert::PolledChannel<Human> pc =
+      sch.CreatePolledChannel<Human>("OutChannel", channel_capacity);
+
+  HumanReverser hr(&pc);
+  pipert::ScheduledChannel<Human> sc2 =
+      sch.CreateScheduledChannel<Human>("ReverserChannel", channel_capacity, nullptr,
+                                        std::bind(&HumanReverser::ReverseName, &hr, std::placeholders::_1));
+
+  HumanPrinter hp(&sc2);
+  pipert::ScheduledChannel<Human> sc1 =
+      sch.CreateScheduledChannel<Human>("PrinterChannel", channel_capacity, nullptr,
+                                        std::bind(&HumanPrinter::PrintName, &hp, std::placeholders::_1));
+
+  sch.Start();
+
+  pipert::Timer::Time time = pipert::Timer::time();
+  for(int i = 0; i < 10; i++) {
+    pipert::PacketToFill<Human> packet_to_fill = sc1.Acquire("Reverser", time, "Jimi Hendrix", 28 + i);
+  }
+
+  usleep(1000);
```

Comment:
The "how-long-should-we-sleep" question also emerged in PR #27, just a heads up.

Reply:
I saw that, thanks. I'm following that topic to get the proper way and replace this solution.

Reply:
See my reply there. I believe that a yield in a loop is the best solution for a test.

Reply:
Added new solution in a fixup.